.github/workflows/deploy.yml

```
@@ -0,0 +1,122 @@
+      run: |
+        cd frontend/web
+        npm ci
+    
+    - name: Build React app
+      run: |
+        cd frontend/web
+        npm run build
+
+    - name: Deploy to GitHub Pages
+      if: github.ref == 'refs/heads/main'
+      uses: peaceiris/actions-gh-pages@v3
```

actionlint (1.7.7)

the runner of "peaceiris/actions-gh-pages@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)